src/storage/mod.rs

```diff
@@ -323,7 +323,7 @@ pub enum MergeError {
 
 #[cfg(feature="serialize")]
 impl<'e, T, D> Storage<'e, T, D>
-    where T: Component + serde::Deserialize,
+    where T: Component + serde::Deserialize<'e>,
```

Comment:
I think we have to lessen the lifetime here and do an `impl<'a, 'e, T, D> &'a Storage<'e, T, D>` because you would have to feed the [`Visitor`](https://docs.serde.rs/serde/de/trait.Visitor.html) with slices that live as long as the borrowed entities.